### PR TITLE
Docs: merge REST API query instructions

### LINKED ISSUE #4453: Docs: remove old content from 'get_started/' and 'restapi/'
This are two structures that were left over from the old documentation and need to be removed, but first we need to check if all the content has been re-assigned in the new documentation. The files considered are:

- [x] [get_started/index.rst](https://aiida-core.readthedocs.io/en/v1.1.0/get_started/index.html) - install plugin  #4468 
- [x] [get_started/index.rst](https://aiida-core.readthedocs.io/en/v1.1.0/get_started/index.html) - develop plugin #4468 
- [x] restapi/index.rst (unlinked in the old docs) - Move "Running the REST API" to "How to share data => AiiDA REST API"
- [x] restapi/index.rst (unlinked in the old docs) - "Serving your data to others" ("Running REST API in production mode" subsection in "Serving data over REST API")
- [x] restapi/index.rst (unlinked in the old docs) - Move "General form of URLs" to REST API Reference section

### Proposition:

~~The first one (`get_started/index.rst`) actually mostly contains infor on installing existing plugins, and could maybe go in a "plugin" section of ["howto: manage your installation"](https://aiida.readthedocs.io/projects/aiida-core/en/latest/howto/installation.html). The second one contains some information on the internals of the RestAPI and should probably be considered for the [corresponding section](https://github.com/aiidateam/aiida-core/issues/4043).~~

See comments

### PR DESCRIPTION
**Pull request by ltalirz**
fixes #4453 

Merge the instructions for querying the REST API into the new "How to
share data" section; moving the details on pagination into the REST API
reference.

Note:  I should probably mention that

 *  I haven't deleted the `Computer` section of the REST API reference - I just moved it down in the list in order to start the list with `Nodes`, which is likely the most important resource by far.
 * in the REST API reference, you will find a couple of `|Group| s` (i.e. an extra space between the class reference and the plural-s). I put this in because `|Group|s` confuses sphinx (with or without the `|...|` substitution syntax) - happy to fix this if there is a fix

### CODE REVIEW

**Comment by giovannipizzi** [comment]
I didn't review this since I see @CasperWA  is assigned, but please remember to add a redirect before  merging

File: docs/source/howto/share_data.rst
Comment (CasperWA) [inline docs/source/howto/share_data.rst:160]:
```suggestion
```

File: docs/source/howto/share_data.rst
Comment (CasperWA) [inline docs/source/howto/share_data.rst:190]: This whole section should be de-indented. Also check the indentation of the parts, I believe the line `Querying the base URL ...` is prepended with one space too many?

File: docs/source/howto/share_data.rst
Comment (CasperWA) [inline docs/source/howto/share_data.rst:185]:
```suggestion
```

File: docs/source/howto/share_data.rst
Comment (CasperWA) [inline docs/source/howto/share_data.rst:183]:
```suggestion
           This also applies to :py:class:`~aiida.orm.users.User` s, which don't have UUIDs (but instead uses email).
```

File: docs/source/reference/rest_api.rst
Comment (CasperWA) [inline docs/source/reference/rest_api.rst:52]: Won't the indentation here make this a comment or similar?

File: docs/source/reference/rest_api.rst
Comment (CasperWA) [inline docs/source/reference/rest_api.rst:100]:
```suggestion
        returns the list of |Node| objects.
        Every node object contains value of attribute called ``pbc1`` if present otherwise null.
```

File: docs/source/reference/rest_api.rst
Comment (CasperWA) [inline docs/source/reference/rest_api.rst:143]:
```suggestion
2. Get a list of all available |Node| types from the database.
```

File: docs/source/reference/rest_api.rst
Comment (CasperWA) [inline docs/source/reference/rest_api.rst:685]:
```suggestion
        returns the list of three |Computer| objects (``limit=3``) starting from the 3rd row (``offset=2``) of the database table.
        The list will be ordered by ascending values of ``id``.
```

File: docs/source/reference/rest_api.rst
Comment (CasperWA) [inline docs/source/reference/rest_api.rst:939]:
```suggestion
    - ``X-Total-Counts`` (custom field): the total number of results returned by the query, i.e. the sum of the results of all pages.
```

File: docs/source/reference/rest_api.rst
Comment (CasperWA) [inline docs/source/reference/rest_api.rst:941]:
```suggestion
    - ``Links``: links to the first, previous, next, and last page. Suppose that you send a request whose results fill 8 pages.
      Then the value of the ``Links`` field would look like::
```

File: docs/source/reference/rest_api.rst
Comment (CasperWA) [inline docs/source/reference/rest_api.rst:974]: I guess we're always "paginating"? :)
```suggestion
    :perpage: How many results to show per page (integer).
```

File: docs/source/reference/rest_api.rst
Comment (CasperWA) [inline docs/source/reference/rest_api.rst:977]:
```suggestion
        Ascending (descending) order for strings corresponds to alphabetical (reverse-alphabetical) order, whereas for datetime objects it corresponds to chronological (reverse-chronological) order.
```

File: docs/source/reference/rest_api.rst
Comment (CasperWA) [inline docs/source/reference/rest_api.rst:999]:
```suggestion
        Pass ``true`` in order to return attributes in the ``/nodes`` endpoint (excluded by default).
```

File: docs/source/reference/rest_api.rst
Comment (CasperWA) [inline docs/source/reference/rest_api.rst:1002]:
```suggestion
        Pass ``true`` in order to return extras in the ``/nodes`` endpoint (excluded by default).
```

File: docs/source/reference/rest_api.rst
Comment (CasperWA) [inline docs/source/reference/rest_api.rst:1186]: I thought we had to specify the complete Python dot-separated path to the classes. And why not use the more easily readable :py:class:`~aiida.orm.users.User`? The tilde (`~`) will make sure only `User` is written in the text.

**Review by CasperWA — CHANGES_REQUESTED** [review]
Cheers @ltalirz.

There are some sections that have been copied over that need a kind touch still I think. I've put in comments to highlight it.
Especially the lists of REST API field descriptions.

Also, personally, I prefer new lines to start with a capital letter, i.e., "Returns" instead of "returns".

File: docs/source/howto/share_data.rst
Comment (ltalirz) [reply docs/source/howto/share_data.rst]: Thanks for noticing, will fix!

File: docs/source/reference/rest_api.rst
Comment (ltalirz) [reply docs/source/reference/rest_api.rst:52]: seems to work fine
image


Admittedly, this is not the most beautiful formatting ever conceived but I think it's good enough to be understandable.

File: docs/source/reference/rest_api.rst
Comment (ltalirz) [reply docs/source/reference/rest_api.rst:1187]: Agreed on both counts - fixed!

**Comment by ltalirz** [comment]
> There are some sections that have been copied over that need a kind touch still I think. I've put in comments to highlight it. Especially the lists of REST API field descriptions.

I guess you mean the tables in the section "Filtering results"?
I agree, I will work on making them look a bit nicer

> Also, personally, I prefer new lines to start with a capital letter, i.e., "Returns" instead of "returns".

Done!

File: docs/source/howto/share_data.rst
Comment (CasperWA) [inline docs/source/howto/share_data.rst:205]:
```suggestion
```

File: docs/source/reference/rest_api.rst
Comment (CasperWA) [inline docs/source/reference/rest_api.rst:101]:
```suggestion
        Every node object contains value of attribute called ``pbc1`` if present otherwise ``null``.
```

File: docs/source/reference/rest_api.rst
Comment (CasperWA) [inline docs/source/reference/rest_api.rst:779]:
```suggestion
        Returns a list of all the |User| objects.
```

File: docs/source/reference/rest_api.rst
Comment (CasperWA) [inline docs/source/reference/rest_api.rst:817]:
```suggestion
        Returns a lists of the |User| objects whose first name starts with ``"aii"``, regardless the case of the characters.
```

File: docs/source/reference/rest_api.rst
Comment (CasperWA) [inline docs/source/reference/rest_api.rst:852]:
```suggestion
        Returns the list of ten |Group| objects (``limit=10``) starting from the 1st row of the database table (``offset=0``) and the list will be ordered by ``user_id`` in descending order.
```

File: docs/source/reference/rest_api.rst
Comment (CasperWA) [inline docs/source/reference/rest_api.rst:896]:
```suggestion
        Returns the details of the |Group| object with ``uuid="a6e5b..."``.
```

File: docs/source/reference/rest_api.rst
Comment (CasperWA) [inline docs/source/reference/rest_api.rst:949]:
```suggestion
```

File: docs/source/reference/rest_api.rst
Comment (CasperWA) [inline docs/source/reference/rest_api.rst:1187]:
```suggestion
.. |Group| replace:: :py:class:`~aiida.orm.groups.Group`
```

**Review by CasperWA — CHANGES_REQUESTED** [review]
Should be good to go after these minor changes.

**Comment by ltalirz** [comment]
Thanks! 

I've added your suggestions + I fixed all tables in the "Filtering results" subsection so that they look nice and are easier to edit in the future + added sub sections to make that section more easy to follow.

File: docs/source/howto/share_data.rst
Comment (CasperWA) [inline docs/source/howto/share_data.rst:166]:
```suggestion
2. The *path* defining the requested *resource*, optionally followed by a more specific *endpoint*.
   For example::
```

File: docs/source/howto/share_data.rst
Comment (CasperWA) [inline docs/source/howto/share_data.rst:184]:
```suggestion
3. (Optional) The *query string* for filtering, ordering and pagination of results.
   For example::
```

File: docs/source/reference/rest_api.rst
Comment (CasperWA) [inline docs/source/reference/rest_api.rst:1004]:
```suggestion
        Example::
```

File: docs/source/reference/rest_api.rst
Comment (CasperWA) [inline docs/source/reference/rest_api.rst:1032]:
```suggestion
```

File: docs/source/reference/rest_api.rst
Comment (CasperWA) [inline docs/source/reference/rest_api.rst:1034]:
```suggestion
```

File: docs/source/reference/rest_api.rst
Comment (CasperWA) [inline docs/source/reference/rest_api.rst:1068]:
```suggestion
      - groups, nodes
```

File: docs/source/reference/rest_api.rst
Comment (CasperWA) [inline docs/source/reference/rest_api.rst:1036]: Could you sort these (and the one above) alphabetically perhaps?

File: docs/source/reference/rest_api.rst
Comment (CasperWA) [inline docs/source/reference/rest_api.rst:1106]:
```suggestion
    - ``node_type="data.remote.RemoteData."`` selects only objects of type :py:class:`~aiida.orm.nodes.data.remote.RemoteData`.
```

File: docs/source/reference/rest_api.rst
Comment (CasperWA) [inline docs/source/reference/rest_api.rst:1105]:
```suggestion
    - ``node_type="data.code.Code."`` selects only objects of type |Code|.
```

File: docs/source/reference/rest_api.rst
Comment (CasperWA) [inline docs/source/reference/rest_api.rst:1111]:
```suggestion
    For example, the request::
```

File: docs/source/reference/rest_api.rst
Comment (CasperWA) [inline docs/source/reference/rest_api.rst:1215]:
```suggestion
```

File: docs/source/reference/rest_api.rst
Comment (CasperWA) [inline docs/source/reference/rest_api.rst:1222]: Maybe sort this list alphabetically as well?

**Review by CasperWA — CHANGES_REQUESTED** [review]

File: docs/source/reference/rest_api.rst
Comment (ltalirz) [reply docs/source/reference/rest_api.rst:1030]: done!

File: docs/source/reference/rest_api.rst
Comment (ltalirz) [reply docs/source/reference/rest_api.rst]: done!

**Review by CasperWA — APPROVED** [review]
One list was still not sorted alphabetically, but I don't want that to hold these changes back, since it's definitely not necessary.

Thanks @ltalirz ! I will squash+merge after merging in `develop`.

**Pull request merged**
